src/drawers/canvasDrawer.ts

```diff
@@ -132,14 +132,14 @@ export function renderPathWithStyle(context: CanvasRenderingContext2D, style: Re
   if (style["stroke"]) {
     context.lineWidth = getStrokeWidth(style);
     const strokeColor = d3.color(style["stroke"]);
-    strokeColor.opacity = getStrokeOpacity(style);
```

Comment:
are we guaranteed that `strokeColor.opacity` will be a valid numerical value? This only works if opacity is a number [0, 1]. But can't it be defined like `"50%"`?

Reply:
I think `stroke-opacity` and `fill-opacity` function the same way as `opacity`, they should just be a float between 0-1:

https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/fill-opacity

hsla/rgba require it to be a float as well:
https://www.w3.org/TR/2011/REC-css3-color-20110607/#hsla-color
https://www.w3.org/TR/2011/REC-css3-color-20110607/#rgba-color

Reply:
Also d3's implementation of color only allows for numerical valued opacity.